mod/groups/start.php

```diff
@@ -1197,6 +1198,31 @@ function _groups_gatekeeper_redirect(\Elgg\Hook $hook) {
 	}
 }
 
+/**
+ * Allow users to visit the group profile page even if group content access mode is set to group members only
+ *
+ * @param \Elgg\Hook $hook Hook
+ * @return void|true
+ */
+function _groups_gatekeeper_allow_profile_page(\Elgg\Hook $hook) {
+
+	$exception = $hook->getValue();
+	if (!$exception instanceof \Elgg\GroupGatekeeperException) {
+		return;
+	}
+	
+	$group = $hook->getEntityParam();
+	if (!$group instanceof ElggGroup) {
+		return;
+	}
+	
+	if (!elgg_in_context('group_profile')) {
```

Comment:
We should move away from context sniffing. Comparing against current route name would be better

Reply:
So compare current_url to group route?

Reply:
Let's of that for now. Will add public API to compare routes later

Reply:
> Will add public API to compare routes later

#11671 for reference

Reply:
changed it